### ISSUE TITLE
Parse comments 

### ISSUE THREAD

**Issue by mohsen1** [typescript-eslint/typescript-eslint#158]
I want to be able to write a rule that warns against unnecessary `@ts-ignore` comments

**What code were you trying to parse?**

```ts
// @ts-ignore
var foo: string = 1
```

**What did you expect to happen?**
Like Babylon or Flow parsers, emit a list of comments next to list of statements 

**What actually happened?**
No comments list is emitted 

**Versions**
Tried in [AST Explorer](https://astexplorer.net/)

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.1.0` |

**Comment by kaicataldo** [comment]
@mohsen1 To clarify, are you asking for a top-level `comments` array? If so, this parser does do that. If you're testing in AST Explorer you need to click the cog and enable `comment`.

**Comment by kaicataldo** [comment]
To clarify, this is only necessary when using the parser directly - ESLint will already enable this flag in its config at parse time.

**Comment by JamesHenry** [comment]
@armano2 ESLint performs the comment attribution

**Comment by kaicataldo** [comment]
@armano2 To be clear, for ESLint's purposes, comment attachment is no longer used and has been deprecated since ESLint v4. Unless there's another use for that feature, I don't think it's needed here. And having worked with comment attachment a lot at this point, I recommend not going down that rabbit hole!

**Comment by armano2** [comment]
yup, i got this information already, as james pointed out https://github.com/eslint/espree/commit/1bcd563d4eb4b4032d2662cc5ccd3bfd841b39d7

**Comment by mohsen1** [comment]
I see `Line` as type of `// comment` lines under `comments` array. When I try to capture those `Line`s in my rule I can not get my callback called:

```js
  create(context) {
    return {
      Line(node) {
      }
   }
  }
```

Should I do something different for comments?
